MQ2GrimGUI.cpp

```diff
@@ -1,505 +1,1510 @@
- */
-PLUGIN_API void OnMacroStart(const char* Name)
+* @fn OnLoadPlugin
+*
+* This is called each time a plugin is loaded (ex: /plugin someplugin), after the
+* plugin has been loaded and any associated -AutoExec.cfg file has been launched.
+* This means it will be executed after the plugin's @ref InitializePlugin callback.
+*
+* This is also called when THIS plugin is loaded, but initialization tasks should
+* still be done in @ref InitializePlugin.
+*
+* @param Name const char* - The name of the plugin that was loaded
+*/
+PLUGIN_API void OnLoadPlugin(const char* Name)
 {
-	// DebugSpewAlways("MQ2GrimGUI::OnMacroStart(%s)", Name);
+	if (!pSpellInspector)
+		pSpellInspector = new grimgui::SpellsInspector();
+
+	if (!pSpellPicker)
+		pSpellPicker = new SpellPicker();
+
+	// check settings file, if logged in use character specific INI else default
+	UpdateSettingFile();
+
+	if (!s_WinSettings.showTitleBars)
+		s_WindowFlags = ImGuiWindowFlags_NoTitleBar;
+
+	if (s_WinSettings.lockAllWin)
+		s_LockAllWin = ImGuiWindowFlags_NoMove;
 }
 
 /**
- * @fn OnMacroStop
- *
- * This is called each time a macro stops (ex: /endmac), after the macro has ended.
- *
- * @param Name const char* - The name of the macro that was stopped.
- */
-PLUGIN_API void OnMacroStop(const char* Name)
+* @fn OnUnloadPlugin
+*
+* This is called each time a plugin is unloaded (ex: /plugin someplugin unload),
+* just prior to the plugin unloading.  This means it will be executed prior to that
+* plugin's @ref ShutdownPlugin callback.
+*
+* This is also called when THIS plugin is unloaded, but shutdown tasks should still
+* be done in @ref ShutdownPlugin.
+*
+* @param Name const char* - The name of the plugin that is to be unloaded
+*/
+PLUGIN_API void OnUnloadPlugin(const char* Name)
 {
-	// DebugSpewAlways("MQ2GrimGUI::OnMacroStop(%s)", Name);
+	DebugSpewAlways("MQ2GrimGUI::OnUnloadPlugin(%s)", Name);
+	RemoveCommand("/grimui");
 }
 
-/**
- * @fn OnLoadPlugin
- *
- * This is called each time a plugin is loaded (ex: /plugin someplugin), after the
- * plugin has been loaded and any associated -AutoExec.cfg file has been launched.
- * This means it will be executed after the plugin's @ref InitializePlugin callback.
- *
- * This is also called when THIS plugin is loaded, but initialization tasks should
- * still be done in @ref InitializePlugin.
- *
- * @param Name const char* - The name of the plugin that was loaded
- */
-PLUGIN_API void OnLoadPlugin(const char* Name)
+PLUGIN_API void InitializePlugin()
 {
-	// update data
-	if (GetGameState() == GAMESTATE_INGAME)
-	{
-		g_CharData.Update();
-		g_TargetData.Update();
-	}
-	// DebugSpewAlways("MQ2GrimGUI::OnLoadPlugin(%s)", Name);
+	DebugSpewAlways("Initializing MQ2GrimGUI");
+	AddCommand("/grimgui", GrimCommandHandler, false, true, false);
+	PrintGrimHelp();
+	pSpellInspector = new grimgui::SpellsInspector();
+	pSpellPicker = new SpellPicker();
+
+	static bool s_DefaultExists = std::filesystem::exists(DEFAULT_INI);
```

Comment:
static is unnecessary. Function is only called once. Need the std::error_code too.